launch_testing/launch_testing/__init__.py

```diff
@@ -0,0 +1,61 @@
+import io
+import re
+from launch.output_handler import LineOutput
+
+
+class InMemoryHandler(LineOutput):
+
+    PREFIXES_TO_FILTER = [
+        b'pid', b'rc',
+        b'RTI Data Distribution Service Evaluation License issued to',
+        b'Expires on']
+
+    def __init__(self, name, ld, expected_lines, regex_match=False):
+        super(LineOutput, self).__init__()
+        self.name = name
+        self.ld = ld
+        self.expected_lines = expected_lines
+        self.killed = False
+        self.left_over_stdout = b''
+        self.left_over_stderr = b''
+        self.stdout_data = io.BytesIO()
+        self.stderr_data = io.BytesIO()
+        self.regex_match = regex_match
+        self.matched = False
+
+    def on_stdout_lines(self, lines):
+        if self.killed:
+            return
+
+        for line in lines.splitlines():
+            # Filter out stdout that comes from underlying DDS implementations
+            if any([line.startswith(prefix) for prefix in self.PREFIXES_TO_FILTER]):
+                continue
+            self.stdout_data.write(line + b'\n')
+
+            # Are we ready to quit?
+            if self.regex_match:
+                self.matched = re.match(self.expected_lines[0], line)
+            else:
+                self.matched = (
+                    len(self.stdout_data.getvalue().splitlines()) >=
+                    len(self.expected_lines))
```

Comment:
Doesn't the line content have to match in order to abort? Currently if the test prints a few unrelated lines the test will abort but the later `check` function will return `False`.